sdk/identity/Azure.Identity/src/MsalConfidentialClient.cs

```diff
@@ -46,20 +46,17 @@ protected override async Task<IConfidentialClientApplication> CreateClientAsync(
 
             if (_certificateProvider != null)
             {
-                X509Certificate2 clientCertificate = await _certificateProvider.GetCertificateAsync(true, default).ConfigureAwait(false);
-
+                X509Certificate2 clientCertificate = await _certificateProvider.GetCertificateAsync(async, cancellationToken).ConfigureAwait(false);
                 confClientBuilder.WithCertificate(clientCertificate);
             }
 
             return confClientBuilder.Build();
-
         }
 
-        public virtual async Task<AuthenticationResult> AcquireTokenForClientAsync(string[] scopes, bool async, CancellationToken cancellationToken)
+        public virtual async ValueTask<AuthenticationResult> AcquireTokenForClientAsync(string[] scopes, bool async, CancellationToken cancellationToken)
         {
-            await EnsureInitializedAsync(async).ConfigureAwait(false);
-
-            return await Client.AcquireTokenForClient(scopes).ExecuteAsync(async, cancellationToken).ConfigureAwait(false);
+            IConfidentialClientApplication client = await GetClientAsync(async, cancellationToken).ConfigureAwait(false);
+            return await client.AcquireTokenForClient(scopes).ExecuteAsync(async, cancellationToken).ConfigureAwait(false);
```

Comment:
Won't this always result in a `Task` being created since it calls MSAL which is always async? If this is the case is anything gained by updating this method to return `ValueTask`?

Reply:
`ValueTask` is generally better for async/sync mixed scenario cause it doesn't create expensive `Task` objects when they aren't needed.